backend/inspirehep/matcher/validators.py

```diff
@@ -22,3 +22,20 @@ def authors_validator(author, result):
     return bool(
         record_author_id_schema_value_pairs & result_author_id_schema_value_pairs
     )
+
+
+def collaboration_validator(author, result):
+    return author.get("collaborations") == get_value(
+        result, "_source.collaborations.value", []
+    )
+
+
+def affiliations_validator(author, result):
+    authors_affiliations = author.get("affiliations", [])
+    result_authors_affiliations = get_value(
+        result, "inner_hits.authors.hits.hits._source.affiliations.value", []
+    )
+    for affiliation_list in result_authors_affiliations:
+        if set(affiliation_list) == set(authors_affiliations):
```

Comment:
Not all affiliations need to match, it's enough if there's one in common
```suggestion
        if set(affiliation_list) & set(authors_affiliations)
```